sdk/core/Azure.Core/tests/ModelReaderWriter/ModelReaderWriterRequestContentTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.ClientModel;
+using System.Net.ClientModel.Core;
+using System.Reflection;
+using Azure.Core.Tests.ModelReaderWriterTests.Models;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests.ModelReaderWriterTests
+{
+    public class ModelReaderWriterRequestContentTests
+    {
+        private const string json = "{\"kind\":\"X\",\"name\":\"Name\",\"xProperty\":100}";
+        private ModelX _modelX;
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            _modelX = ModelReaderWriter.Read<ModelX>(BinaryData.FromString(json));
+        }
+
+        [Test]
+        public void CanCalculateLength()
+        {
+            //use IModelSerializable
+            var content = RequestContent.Create(PipelineContent.CreateContent((IModel<ModelX>)_modelX));
```

Comment:
What is the reason that we need a RequestContent here instead of a PipelineContent?